tests/rptest/tests/random_node_operations_test.py

```diff
@@ -156,12 +156,24 @@ def _start_redpanda(self, mixed_versions, with_tiered_storage):
                                    self.previous_version)
 
             self.redpanda.set_seed_servers(self.nodes_with_prev_version)
-            self.redpanda.start(auto_assign_node_id=True,
+
+            # Installing a license is required for version upgrades with enterprise features
+            # We need to install it before the new nodes try to join the cluster, so install
+            # it after the old nodes are started but before the new ones are starting
+            self.redpanda.start(nodes=self.redpanda.nodes[:with_prev_version],
+                                auto_assign_node_id=True,
+                                omit_seeds_on_idx_one=False)
+            self.redpanda.install_license()
```

Comment:
Side effect of all this is that these tests will now fail locally unless a license is pushed in through the environment, right?

Reply:
That's correct. (I've added this note to the commit message now)